touch/src/main.rs

```diff
@@ -0,0 +1,240 @@
+use chrono::NaiveDateTime;
+use clap::{load_yaml, App, ArgMatches};
+use filetime::{set_file_atime, set_file_mtime, FileTime};
+use std::fs::{metadata, File};
+use std::io::Result;
+use std::process;
+use std::time::SystemTime;
+
+fn main() {
+    let yaml = load_yaml!("touch.yml");
+    let matches = App::from_yaml(yaml).get_matches();
+
+    // get files list in argument
+    let files = if matches.is_present("FILE") {
+        matches.values_of("FILE").unwrap().collect::<Vec<_>>()
+    } else {
+        Vec::new()
+    };
+
+    if files.len() > 0 {
+        touch(&matches, &files).unwrap();
+    } else {
+        eprintln!("touch: a file name is required.\n");
+        process::exit(1);
+    }
+}
+
+fn touch(matches: &ArgMatches, files: &Vec<&str>) -> Result<()> {
+    for filename in files {
+        // if file already exist in the current directory
+        let file_metadata = metadata(&filename);
+        let no_create = !matches.is_present("nocreate");
+        if file_metadata.is_err() && no_create {
+            match File::create(&filename) {
+                Ok(_) => (),
+                Err(e) => eprintln!("touch: Failed to create file {} : {}", &filename, e),
+            }
+        } else {
+            let date = matches.is_present("date");
+            if date {
+                let date_value = matches.value_of("date").unwrap();
+                let native_date =
+                    NaiveDateTime::parse_from_str(&date_value, "%Y-%m-%d %H:%M:%S").unwrap();
+                let newfile_time = FileTime::from_unix_time(
+                    native_date.timestamp(),
+                    native_date.timestamp_subsec_millis(),
+                );
+                update_time(matches, &filename, newfile_time);
+            } else {
+                let newfile_time = FileTime::from_system_time(SystemTime::now());
+                update_time(matches, &filename, newfile_time);
+            }
+        }
+    }
+    Ok(())
+}
+
+fn update_time(matches: &ArgMatches, path: &str, filetime: FileTime) {
+    let access_time = matches.is_present("accesstime");
+    let modification = matches.is_present("modification");
+    let time = matches.is_present("time");
+    let time_value = matches.value_of("time");
```

Comment:
I think it would be better to do the following
```rust
let time = matches.is_present("time");
let time_value = if time {
    matches.value_of("time").unwrap()
} else {
    ""
}
```

that way your next if statement would be more clean

Reply:
is there a shorter way to handle value ?
like in Javascript I can use ternary operator

Reply:
This could also be:
```rust
let time = matches.value_of("time").unwrap_or("");
```

Reply:
Yes I prefer using `unwrap_or` much better for reading